crates/integration-tests/tests/federation/basic/skip_include.rs

```diff
@@ -48,14 +48,8 @@ fn skip_include() {
     {
       "data": {
         "pullRequestsAndIssues": [
-          {
-            "author": {}
-          },
-          {
-            "author": {
-              "id": "123"
-            }
-          },
+          {},
+          {},
```

Comment:
Interesting, is it an arbitrary choice or is the behaviour of having both skip and include specified?

Reply:
It's the type condition that wasn't tracked correctly, the only thing left after skip/include is:
```
                    fragment IssueFragment on Issue {
                        author {
                            ...AuthorFragment
                        }
                    }
```
and the first two items weren't `Issue`s but `PullRequest`

Reply:
I was merging fields too aggressively which led me to "forget" that type conditions applied